lti_consumer/lti_1p3/extensions/rest_framework/serializers.py

```diff
@@ -227,3 +230,90 @@ class Meta:
             'resultMaximum',
             'comment',
         )
+
+
+# pylint: disable=abstract-method
+class LtiContextSerializer(serializers.Serializer):
+    """
+    Serializer for a LTI Context
+    """
+    id = UsageKeyField()
+
+
+class LtiNrpsContextMemberBasicSerializer(serializers.Serializer):
+    """
+    Non PII fields serializer for Context Member.
+    """
+    status = serializers.SerializerMethodField()
+    user_id = serializers.SerializerMethodField()
+    roles = serializers.SerializerMethodField()
+
+    def get_user_id(self, user):
+        """
+        Get external user id for the user.
+        """
+        externalid, _ = compat.get_or_create_externalid(user)
```

Comment:
Calling this here will not be optimal.

Ideally, the external ID should be prefetched/annotated in the queryset, and mapped using the `origin` parameter. I haven't tested this out to see if it works though.

Reply:
I will be using the batch external id to create PR. But for now, as we have opened two seperate PR on ``edx-platform``, this PR depends on one of them. I will remove this to use the batch method when ``edx-platform`` PR related to this gets updated.